### PR TITLE
DEPR: Enforce deprecation of include_groups in groupby.apply

### PR DESCRIPTION
**Pull request by rhshadrach**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

The PR promoting this to a FutureWarning was merged straight into 2.3.x instead of main.

https://github.com/pandas-dev/pandas/pull/59751

### CODE REVIEW

File: pandas/core/groupby/groupby.py [hunk 1]
```diff
@@ -1419,18 +1419,17 @@ def apply(self, func, *args, include_groups: bool = True, **kwargs) -> NDFrameT:
...
+        include_groups : bool, default False
             When True, will attempt to apply ``func`` to the groupings in
             the case that they are columns of the DataFrame. If this raises a
             TypeError, the result will be computed with the groupings excluded.
             When False, the groupings will be excluded when applying ``func``.
 
             .. versionadded:: 2.2.0
 
-            .. deprecated:: 2.2.0
+            .. versionchanged:: 3.0.0
```
Comment (mroeschke) [inline pandas/core/groupby/groupby.py:1430]: Should we also deprecate this keyword in 3.0 or in a future minor release?

File: pandas/core/resample.py [hunk 2]
```diff
@@ -2729,16 +2721,6 @@ def _asfreq_compat(index: FreqIndexT, freq) -> FreqIndexT:
...
-    # GH#7155 - rewrite warning to appear as if it came from `.resample`
-    target_message = "DataFrameGroupBy.apply operated on the grouping columns"
-    new_message = _apply_groupings_depr.format("DataFrameGroupBy", "resample")
-    with rewrite_warning(
-        target_message=target_message,
-        target_category=DeprecationWarning,
-        new_message=new_message,
-    ):
-        result = grouped.apply(how, *args, include_groups=include_groups, **kwargs)
+def _apply(grouped: GroupBy, how: Callable, *args, **kwargs) -> DataFrame:
```
Comment (mroeschke) [inline pandas/core/resample.py:2724]: Could we just inline this function?

File: pandas/core/groupby/groupby.py (hunk 1 above)
Comment (rhshadrach) [reply pandas/core/groupby/groupby.py:1430]: I was thinking to add the deprecation for 3.1.

File: pandas/core/resample.py (hunk 2 above)
Comment (rhshadrach) [reply pandas/core/resample.py]: Yup - good call.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @rhshadrach

**Pull request merged**